### ISSUE TITLE
Overriding the volume mounted?

### ISSUE THREAD

**Issue by leeeennyy** [microsoft/vscode-docker#1206]
Hi,

I have multiple projects in my solution, where a project will be dependent on another project. For example, ProjectB is dependent on ProjectA.

```
solutionFolder
|-- ProjectA
    |-- ProjectA.csproj
|-- ProjectB
    |-- ProjectB.csproj
|-- solution.sln
```

I have setup my `launch.json` to build ProjectB, however it only mounts the files from the ProjectB folder instead of the server folder. Is there a way where I can mount it under the `solutionFolder` instead so that the files required in ProjectA are debuggable in the container as well?

launch.json for reference
```
{
    "name": "Docker: ProjectB",
    "type": "docker-coreclr",
    "request": "launch",
    "preLaunchTask": "build-projectb",
    "appProject": "${workspaceFolder}/ProjectB/ProjectB.csproj",
    "dockerBuild": {
        "dockerfile": "${workspaceFolder}/ProjectB",
        "context": "${workspaceFolder}",
        "tag": "projectb:dev"
    },
    "dockerRun": {
        "containerName": "projectb",
        "ports": [
            {
                "hostPort": "5000",
                "containerPort": "5000"
            }
        ]
    }
}
```

**Comment by bwateratmsft** [comment]
Yes, you can add volumes under the "dockerRun" option. Most likely, your Dockerfile will need some changes as well. We're also working on supporting this more directly.

**Comment by bwateratmsft** [comment]
Ah, I see what you mean. It would be ugly in the project files, but you can possibly do something like this:
```
"volumes": {
    "localPath": "${workspaceFolder}",
    "containerPath": "/src",
    "permissions": "ro"
}
```

In ProjectB.csproj:
```
...
<ItemGroup>
  <ProjectReference Include="..\ProjectA\ProjectA.csproj" Condition="Exists('..\ProjectA\ProjectA.csproj')" />
  <ProjectReference Include="/src/ProjectA/ProjectA.csproj" Condition="Exists('/src/ProjectA/ProjectA.csproj')" />
</ItemGroup>
...
```

That being said, for debugging the build doesn't happen within the container, it happens on the host, and the output is mapped into /app. Since ProjectA.dll is present in ProjectB's output folder, debugging worked without needing the above project file change--all I needed was the Dockerfile in the solution folder, with "dockerfile" and "context" specified in "dockerBuild", as you have in the original post; along with appProject.

**Comment by bwateratmsft** [comment]
Resolved in #1245. We allow any user-specified volumes.
